src/auth/authenticationManager.ts

```diff
@@ -100,3 +104,61 @@ export const useCheckIsLoggedIn = () => {
 
   return { isLoading, isSuccess };
 };
+
+export function decodeToken(token: string) {
+  const base64Url = token.split('.')[1];
+  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  const jsonPayload = decodeURIComponent(
+    atob(base64)
+      .split('')
+      .map(function (c) {
+        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+      })
+      .join(''),
+  );
+
+  const { exp } = JSON.parse(jsonPayload);
+  return exp;
+}
+
+export const isTokenExpired = (decodeToken: number) => {
+  const timeMsUntilTokenExpires = 5 * 60 * 1000;
+  const isTimeUp = decodeToken * 1000 - timeMsUntilTokenExpires;
+  const expired = Date.now() >= decodeToken * 1000 || Date.now() >= isTimeUp;
+
+  return expired;
+};
+
+export function useIsIdTokenExpired() {
+  let currentIdToken = sessionStorage.getItem('IdToken');
```

Comment:
use `const` if your variable is not mutated

Reply:
changed